src/neptune_scale/api/attribute.py

```diff
@@ -211,25 +191,41 @@ def cleanup_path(path: str) -> str:
     Traceback (most recent call last):
     ...
     ValueError: Invalid path: `a//b/c`. Path components must not be empty.
+    >>> cleanup_path('a/ /b/c')
+    Traceback (most recent call last):
+    ...
+    ValueError: Invalid path: `a/ /b/c`. Path components must not be empty.
+    >>> cleanup_path('a/b/c ')
+    Traceback (most recent call last):
+    ...
+    ValueError: Invalid path: `a/b/c `. Path cannot contain leading or trailing whitespace.
     """
 
-    path = path.strip()
-    if path in ("", "/"):
+    if path.strip() in ("", "/"):
```

Comment:
This is a major change though. Since you're not assigning the stripped path back to path, if the path has some whitespace, it will stay and break all the instructions below.

E.g.
previously " /path/to/sth/ " -> strip() -> "/path/to/sth/" -> path[1:] -> "path/to/sth/" -> error because it ends with /
now " /path/to/sth/ " -> strip() is not saved -> " /path/to/sth/ " -> path[1:] ignored bc path starts with " " -> " /path/to/sth/ " -> no error because it ends with " "...

Reply:
please add tests